packages/relay/src/lib/services/clientService/index.ts

```diff
@@ -0,0 +1,276 @@
+      .inc(1);
+
+    const hederaNetwork: string = (process.env.HEDERA_NETWORK || '{}').toLowerCase();
+    this.clientMain = this.initClient(this.logger, hederaNetwork);
+    this.client = this.initSDKClient(this.logger, this.register);
+
+    this.resetCounters();
+  }
+
+  /**
+   * Reset all counters with predefined configuration.
+   */
+  private resetCounters() {
+    this.transactionCount = parseInt(process.env.CLIENT_TRANSACTION_RESET!) || 0;
+    this.resetDuration = Date.now() + parseInt(process.env.CLIENT_DURATION_RESET!) || 0;
+    this.errorCount = parseInt(process.env.CLIENT_ERROR_RESET!) || 0;
+
+    this.shouldReset = false;
+  }
+
+  /**
+   * Configure SDK Client from main client
+   * @param {Logger} logger
+   * @param {Registry} register
+   * @returns SDK Client
+   */
+  private initSDKClient(logger: Logger, register: Registry): SDKClient {
+    return new SDKClient(this.clientMain, logger.child({ name: `consensus-node` }), register);
+  }
+
+  /**
+   * Configure Client
+   * @param {Logger} logger
+   * @param {string} hederaNetwork
+   * @param {string | null} type
+   * @returns Client
+   */
+  private initClient(logger: Logger, hederaNetwork: string, type: string | null = null): Client {
+    let client: Client;
+    if (hederaNetwork in this.chainIds) {
+      client = Client.forName(hederaNetwork);
+    } else {
+      client = Client.forNetwork(JSON.parse(hederaNetwork));
+    }
+
+    if (type === 'eth_sendRawTransaction') {
+      if (process.env.OPERATOR_ID_ETH_SENDRAWTRANSACTION && process.env.OPERATOR_KEY_ETH_SENDRAWTRANSACTION) {
+        client = client.setOperator(
+          AccountId.fromString(process.env.OPERATOR_ID_ETH_SENDRAWTRANSACTION),
+          PrivateKey.fromString(process.env.OPERATOR_KEY_ETH_SENDRAWTRANSACTION)
+        );
+      } else {
+        logger.warn(`Invalid 'ETH_SENDRAWTRANSACTION' env variables provided`);
+      }
+    } else {
+      if (process.env.OPERATOR_ID_MAIN && process.env.OPERATOR_KEY_MAIN) {
+        client = client.setOperator(
+          AccountId.fromString(process.env.OPERATOR_ID_MAIN.trim()),
+          PrivateKey.fromString(process.env.OPERATOR_KEY_MAIN)
+        );
+      } else {
+        logger.warn(`Invalid 'OPERATOR' env variables provided`);
+      }
+    }
+
+    client.setTransportSecurity(process.env.CLIENT_TRANSPORT_SECURITY === 'true' || false);
+    client.setRequestTimeout(parseInt(process.env.SDK_REQUEST_TIMEOUT || '10000'));
+
+    logger.info(
+      `SDK client successfully configured to ${JSON.stringify(hederaNetwork)} for account ${
+        client.operatorAccountId
+      } with request timeout value: ${process.env.SDK_REQUEST_TIMEOUT}`
+    );
+
+    return client;
+  }
+
+  /**
+   * Return main client
+   * @returns Main Client
+   */
+  public getMainClient() {
+    return this.clientMain;
+  }
+
+  /**
+   * Return configured sdk client
+   * @returns SDK Client
+   */
+  public getSDKClient(): SDKClient {
```

Comment:
Should have a more informative function name.
Thinking on it myself but something suggests the client is managed in instances.

Reply:
You're right, although there is a option to disable this whole reinitialisation feature and it's a valid name then. I'm open to suggestions

Reply:
Hmm good point.
Okay how about this, leave this as `getSDKClient()` and change the other to `getSingletonSDKClient()`.
SO maybe the emphasis is on the fact that the other is the instance that is long lived across the application.

Reply:
Renamed.